packages/merge-on-green/src/merge-logic.ts

```diff
@@ -141,166 +172,165 @@ mergeOnGreen.hasMOGLabel = async function hasMOGLabel(
+    console.info(
+      `called getStatusi in ${Date.now() - start}ms ${owner}/${repo}, ${
+        data[0].context
+      }`
+    );
+    return data;
   } catch (err) {
-    console.error(err);
     return [];
   }
 };
 
-mergeOnGreen.getBranchProtection = async function getBranchProtection(
+/**
+ * Function iterates through the multiple pages of check statuses and concatenates them into a large array
+ * @param owner of pr (from Watch PR)
+ * @param repo of pr (from Watch PR)
+ * @param github unique installation id for each function
+ * @param headSha the head sha commit
+ * @returns an array of Check Statuses that has their names and statuses
+ */
+mergeOnGreen.iterateGetStatusi = async function iterateGetStatusi(
   owner: string,
   repo: string,
-  github: GitHubAPI
-) {
-  const branchProtection = (
-    await github.repos.getBranchProtection({
-      owner,
-      repo,
-      branch: 'master',
-    })
-  ).data.required_status_checks.contexts;
-  console.log(`branch protection for ${owner}/${repo}: ${branchProtection}`);
-  return branchProtection;
-};
-
-mergeOnGreen.getRequiredChecks = async function getRequiredChecks(
   github: GitHubAPI,
-  owner: string,
-  repo: string
-): Promise<string[]> {
-  const [checksByLanguage, languageMap] = await Promise.all([
-    mergeOnGreen.requiredChecksByLanguage(github),
-    mergeOnGreen.mapReposToLanguage(github),
-  ]);
-  if (checksByLanguage && languageMap) {
-    const language = languageMap.find(
-      (element: Language) => element.repo === `${owner}/${repo}`
-    );
-    if (language !== undefined) {
-      if (checksByLanguage[language.language].repoOverrides !== undefined) {
-        const isOverriden = checksByLanguage[
-          language.language
-        ].repoOverrides.find(
-          (element: RepoOverrides) => element.repo === `${owner}/${repo}`
-        );
-        if (isOverriden) {
-          console.log(
-            `Your language's required checks were overridden because of the repo ${owner}/${repo}` +
-              `it is ${isOverriden.useBranchProtectionRules} that ${owner}/${repo} uses default branch protection`
-          );
-          if (isOverriden.useBranchProtectionRules === true) {
-            console.log(`Checking branch protection for ${owner}/${repo}`);
-            const branchProtection = await mergeOnGreen.getBranchProtection(
-              owner,
-              repo,
-              github
-            );
-            console.log(`Using native branch protection for ${owner}/${repo}`);
-            return branchProtection;
-          }
-          return isOverriden.requiredStatusChecks;
-        }
-      }
-      return checksByLanguage[language.language].requiredStatusChecks;
-    } else {
-      console.info(
-        `${owner}/${repo} does not have a corresponding language in sloth/repos.json`
-      );
-      return [];
+  headSha: string
+): Promise<CheckStatus[]> {
+  let results: CheckStatus[] = [];
+  for (let i = 0; i < 10; i++) {
```

Comment:
"10" is an oddly round number.  What's happening here?  Is that a concurrency limit or something?